sale_margin_percentage/tests/test_sale_order.py

```diff
@@ -80,8 +80,9 @@ def test_07_margin_total(self):
         self.create_so_line(sale, product=self.product_cost_5, quantity=4, price=13)
         self.create_so_line(sale, product=self.product_cost_10, quantity=5, price=12)
         self.assertEqual(sale.order_line.mapped("margin"), [48.0, -60.0, -12.0, 8.0, 0.0, 0.0, 32.0, 10.0])
-        self.assertEqual(
-            sale.order_line.mapped("margin_percentage"), [100.0, -100.0, -25.0, 16.67, 0.0, 0.0, 61.54, 16.67]
-        )
+        calculated_margins = sale.order_line.mapped("margin_percent")
+        expected_margins = [1.0, -1.0, -0.25, 0.1667, 0.0, 0.0, 0.6154, 0.1667]
+        for expected, calculated in zip(expected_margins, calculated_margins):
```

Comment:
Why?

We should avoid loops within tests

Reply:
Because if I use `assertEqual` I got error since they are not computational equal and the `assertAlmostEqual` don't support list, the `numpy.testing` have a `assertAlmostEqual` that support list but I don't think use numpy is better than the loop, what do you think?

Reply:
I think you could round the computed value

Or even better, use `assertRecordValues` that will take care of rounding.

Reply:
done, the `assertRecordValues`didn't take care of rounding so I rounded the computed value.